assert-instr/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+#![feature(proc_macro)]
+
+extern crate assert_instr_macro;
+extern crate backtrace;
+#[macro_use]
+extern crate lazy_static;
+
+use std::collections::HashMap;
+use std::env;
+use std::process::Command;
+use std::str;
+
+pub use assert_instr_macro::*;
+
+lazy_static! {
+    static ref DISASSEMBLY: HashMap<String, Vec<Function>> = disassemble_myself();
+}
+
+struct Function {
+    instrs: Vec<Instruction>,
+}
+
+struct Instruction {
+    parts: Vec<String>,
+}
+
+fn disassemble_myself() -> HashMap<String, Vec<Function>> {
+    let me = env::current_exe().expect("failed to get current exe");
+    let output = Command::new("objdump")
```

Comment:
How portable is this? Does this work on Windows? Or only Linux/MacOS?

Reply:
An excellent question! AFAIK there's *some* method of doing this on all platforms. This'll require some degree of platform-specific code, but it's something I can take care of.